src/parsley/fields.py

```
@@ -227,6 +227,22 @@ def decode(self, data: bytes) -> str:
+            encoded_data = int(value, 0).to_bytes((self.length + 7) // 8, byteorder='big')
+            return (encoded_data, self.length)
+
+        # Named bitfield where value is a string of flag names seperated by "|"
+        # Default string represents "no bits set" (bitmask == 0)
+        if value == self.default:
+            bitfield_value = 0
+        else:
+            bitfield_value = 0
+            for name in value.split('|'):
+                if name not in self.map_name_offset:
+                    raise ValueError(f'Name "{name}" not found in bitfield "{self.name}"')
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)